Assets/Scripts/Player.cs

```diff
@@ -43,6 +45,8 @@ public class Player : MonoBehaviour
             if (MoveSprite(nextTileX, nextTileY, 'P', direction)) {
                 grid.CheckHeteroMatching();
                 grid.CheckWin();
+                grid.AddState();
+                grid.removeFirstState = true;
```

Comment:
maybe this could be a parameter of the `AddState()` method?
since it looks like it only affects that method it woud be clearer that it is related to its behavior